haste/windowed_conformal_model/conformal_interestingness_model.py

```diff
@@ -32,6 +32,28 @@ def __init(self):
         # TODO: create mongo client? (or pass is collection passed in?)
         pass
 
+    def queryMongo(mongo_collection, substream_id):
+
+    #Search for documents with specific substream_id having image_point_number = 1 and color_channel = 2 (Green)
+    cursor = mongo_collection.find({'substream_id': substream_id, 'metadata.imaging_point_number': 1,
+                                    'metadata.color_channel': 2}).sort('timestamp', 1)
+
+    correlation = []
+    sum_of_intensities = []
+    x_time = []
+
+    for document in cursor:
```

Comment:
Consider doing like this?
https://stackoverflow.com/questions/8723613/pymongo-and-toarray-method

Reply:
Also, consider using the projection argument, see: https://api.mongodb.com/python/current/api/pymongo/collection.html#pymongo.collection.Collection.find

Reply:
I don't really see the benefit since I need to go through each document anyway to find the value of the feature, right?

Reply:
Yes, no worries.
But perhaps use the projection argument to only fetch the values you need (this will make the query faster since we're getting so many fields) Have a look at my query on the master branch in this file.

Reply:
fixed!